consul/kv_consul.go

```
@@ -271,8 +271,10 @@ func (kv *consulKV) Create(
 			if ok && err == nil {
 				return kvPair, err
 			}
-			kv.client.Session().Destroy(sessionPair.Session, nil)
-			kv.Delete(key)
+			_, _ = kv.client.Session().Destroy(sessionPair.Session, nil)
+			if _, err := kv.Delete(key); err != nil {
+				return nil, err
```

I think you still want to return the error from `kv.client.KV().Acquire(`